File: src/main/java/architecture/goldenboughs_lib/init/tag/LibBlockTags.java
package architecture.goldenboughs_lib.init.tag;

import architecture.goldenboughs_lib.core.GoldenBoughsLib;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.tags.BlockTags;
import net.minecraft.tags.TagKey;
import net.minecraft.world.level.block.Block;

public final class LibBlockTags {

	private static TagKey<Block> createTag(String name) {
		return createTag(GoldenBoughsLib.modRl(name));
	}

	private static TagKey<Block> createTag(ResourceLocation location) {
		return BlockTags.create(location);
	}

	private static TagKey<Block> createCTag(String name) {
		return createTag(ResourceLocation.fromNamespaceAndPath("c", name));
	}

	private static TagKey<Block> createMcTag(String name) {
		return createTag(ResourceLocation.withDefaultNamespace(name));
	}
}

File: src/main/kotlin/architecture/goldenboughs_lib/init/tag/LibBlockTags.kt
package architecture.goldenboughs_lib.init.tag

import architecture.goldenboughs_lib.util.LibUtil.modRl
import architecture.goldenboughs_lib.util.LibUtil.rlOf
import net.minecraft.resources.ResourceLocation
import net.minecraft.tags.BlockTags
import net.minecraft.tags.TagKey
import net.minecraft.world.level.block.Block

object LibBlockTags {
	private fun createTag(name: String): TagKey<Block?> {
		return createTag(modRl(name))
	}

	private fun createTag(location: ResourceLocation): TagKey<Block?> {
		return BlockTags.create(location)
	}

	private fun createCTag(name: String): TagKey<Block?> {
		return createTag(rlOf("c", name))
	}

	private fun createMcTag(name: String): TagKey<Block?> {
		return createTag(ResourceLocation.withDefaultNamespace(name))
	}
}
